src/constructor.io/client/modules/Catalog.cs

```diff
@@ -971,5 +971,260 @@ public async Task<SearchabilitiesResponse> PatchSearchabilities(PatchSearchabili
+        /// </summary>
+        /// <param name="retrieveSortOptionBySortByRequest">Constructorio's <see cref="RetrieveSortOptionBySortByRequest"/> object model.</param>
+        /// <returns>Constructorio's <see cref="SortOptionList"/> object.</returns>
+        public async Task<SortOptionList> RetrieveSortOptionsBySortBy(RetrieveSortOptionBySortByRequest retrieveSortOptionBySortByRequest)
+        {
+            string url = CreateSortOptionsUrl(retrieveSortOptionBySortByRequest, retrieveSortOptionBySortByRequest.FilterBySortBy);
+            Dictionary<string, string> requestHeaders = new Dictionary<string, string>();
+            AddAuthHeaders(this.Options, requestHeaders);
+
+            string result;
+            try
+            {
+                result = await MakeHttpRequest(this.Options, HttpMethod.Get, url, requestHeaders);
+            }
+            catch (Exception e)
+            {
+                throw new ConstructorException(e);
+            }
+
+            if (result != null)
+            {
+                return JsonConvert.DeserializeObject<SortOptionList>(result);
+            }
+
+            throw new ConstructorException("RetrieveSortOptionsBySortBy response data is malformed");
+        }
+
+        /// <summary>
+        /// Replaces all existing Sort Options with a new list of Sort Options.
+        /// At most 50 Sort Options can be provided.
+        /// </summary>
+        /// <param name="sortOptionsListRequest">Constructorio's <see cref="SortOptionsListRequest"/> object model.</param>
+        /// <returns>Constructorio's <see cref="SortOptionList"/> object.</returns>
+        public async Task<SortOptionList> SetSortOptions(SortOptionsListRequest sortOptionsListRequest)
+        {
+            string url = CreateSortOptionsUrl(sortOptionsListRequest);
+            Dictionary<string, string> requestHeaders = new Dictionary<string, string>();
+            AddAuthHeaders(this.Options, requestHeaders);
+
+            string result;
+            try
+            {
+                result = await MakeHttpRequest(this.Options, HttpMethod.Put, url, requestHeaders, sortOptionsListRequest.GetSortOptionsList());
+            }
+            catch (Exception e)
+            {
+                throw new ConstructorException(e);
+            }
+
+            if (result != null)
+            {
+                return JsonConvert.DeserializeObject<SortOptionList>(result);
+            }
+
+            throw new ConstructorException("SetSortOptions response data is malformed");
+        }
+
+        /// <summary>
+        /// Deletes a list of Sort Options, identified by sort_by and sort_order.
+        /// </summary>
+        /// <param name="sortOptionsListRequest">Constructorio's <see cref="SortOptionsListRequest"/> object model.</param>
+        /// <returns>True if delete request succeeds.</returns>
+        public async Task<bool> DeleteSortOption(SortOptionsListRequest sortOptionsListRequest)
```

Comment:
Should this be `DeleteSortOptions`(plural)?

And similar to the previous comments, should we just simplify this to just accept two arguments? (list of sort options to remove and section)

Reply:
Ooh nice catch on the missing `s`. As for the simplification, we're running with the Request Object pattern. See [Slack Thread Here](https://constructor.slack.com/archives/C02RSEE9QE4/p1683741624059059).